github/create_deployment.yaml

```diff
@@ -0,0 +1,111 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-create-deployment
+  description: |
+    This Task will create a GitHub deployment for a repository.
+spec:
+  params:
+    - name: GITHUB_HOST_URL
+      description: |
+        The GitHub host, adjust this if you run a GitHub enterprise.
+      default: "api.github.com"
+    - name: REPO_FULL_NAME
+      description: |
+        The GitHub repository full name, i.e: tektoncd/catalog.
+    - name: REF
+      description: |
+        The ref to deploy. This can be a branch, tag, or SHA.
+    - name: TASK
+      description: |
+        Specifies a task to execute (e.g., deploy or deploy:migrations).
+      default: "deploy"
```

Comment:
I noticed that the other github tasks don't really provide default values, but in the contributor's guidelines [it recommends to provide as many default values](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md#technical-requirements). I'm just falling back to the GitHub API's default values for most of these.